InferSharp - analysis

steinho@4963ADB:~/infersharp$ ./run_infersharp.sh ../Repos/assessments-frontend/Assessments.Frontend.Web/bin/Debug/net5.0/
Processing {../Repos/assessments-frontend/Assessments.Frontend.Web/bin/Debug/net5.0/}
Copying binaries to a staging folder...\n
Code translation started...
Translation stage 1/3: Loading binaries.
Translation stage 2/3: Computing type environment.
Translation stage 3/3: Computing control-flow graph.

Coverage Statistics:

Method successfully translated: 2847 (91%)
Method partially translated: 285 (9%)
Instructions translated: 42738 (44%)
Instructions skipped: 55137 (56%)
======================================

Code translation completed. Analyzing...


/home/steinho/Repos/assessments-frontend/Assessments.Frontend.Web/Views/Redlist/Species/2021/List/partials/_ListView.cshtml:59: error: Null Dereference
  pointer `returned by get_Category()` could be null and is dereferenced at line 59, column 1.
  57.
  58.                     <span class="list_cell categorytag">
  59.                         <span class="@item.Category.Substring(0,2) category_tag">
      ^
  60.                             <span class="category_tag_label">@item.Category</span>
  61.                         </span>

/home/steinho/Repos/assessments-frontend/Assessments.Mapping/Helpers/CategoryAndCriteriaHelper.cs:368: error: Null Dereference
  pointer `%1` could be null and is dereferenced at line 368, column 17.
  366.
  367.                 //Console.WriteLine("klassifisering: " + overordnetKlassifisering);
  368.                 total.Kategori = overordnetKlassifisering.Substring(overordnetKlassifisering.Length - 2);
                       ^
  369.                 total.Kriterie = "";
  370.             }


Found 2 issues
          Issue Type(ISSUED_TYPE_ID): #
  Null Dereference(NULL_DEREFERENCE): 2

--- a/Assessments.Frontend.Web/Infrastructure/DataRepository.cs
+++ b/Assessments.Frontend.Web/Infrastructure/DataRepository.cs
@@ -51,7 +51,9 @@ namespace Assessments.Frontend.Web.Infrastructure
                 }
                 else // download file
                 {
-                    var blob = new BlobContainerClient(_configuration["ConnectionStrings:AzureBlobStorage"], "assessments").GetBlobClient(name);
+                    string connectionString = _configuration["ConnectionStrings:AzureBlobStorage"];
+                    if (string.IsNullOrWhiteSpace(connectionString)) throw new System.Exception("Missing required config for azure blog storage: ConnectionStrings:AzureBlobStorage");
+                    var blob = new BlobContainerClient(connectionString, "assessments").GetBlobClient(name);
                     var response = await blob.DownloadContentAsync();
 
                     fileContent = response.Value.Content.ToString();

--- a/Assessments.Mapping/Helpers/CategoryAndCriteriaHelper.cs
+++ b/Assessments.Mapping/Helpers/CategoryAndCriteriaHelper.cs
@@ -363,9 +363,13 @@ namespace Assessments.Mapping.Helpers
                 {
                     Console.WriteLine("mangler overordnetKlassifisering");
                 }
+                else if (overordnetKlassifisering.Length > 1)
+                {
+                    total.Kategori = overordnetKlassifisering.Substring(overordnetKlassifisering.Length - 2);
+                }
 
                 //Console.WriteLine("klassifisering: " + overordnetKlassifisering);
-                total.Kategori = overordnetKlassifisering.Substring(overordnetKlassifisering.Length - 2);
+                
                 total.Kriterie = "";
             }
             return total;

